examples/task-lifecycle.py

```diff
@@ -0,0 +1,264 @@
+import asyncio
+import random
+import itertools
+import time
+
+
+from avocado.utils.astring import tabular_output
+
+
+DEBUG = False
+
+
+def debug(msg):
+    if DEBUG:
+        print(msg)
+
+
+async def sleep_random():
+    await asyncio.sleep(random.random())
+
+
+def true_or_false(handicap=3):
+    """Returns a random positive or negative outcome, with some bias."""
+    if handicap > 1:
+        choices = [True] + ([False] * handicap)
+    else:
+        choices = [False] + ([True] * abs(handicap))
+    return random.choice(choices)
+
+
+def mock_check_task_requirement():
+    # More success than failures, please
+    return true_or_false(-8)
+
+
+def mock_check_task_start():
+    # More success than failures, please
+    return true_or_false(-6)
+
+
+def mock_monitor_task_finished():
+    # More failures than successes, please
+    return true_or_false(5)
+
+
+class Task:
+    """Used here as a placeholder for an avocado.core.nrunner.Task."""
+    def __init__(self, identification):
+        self._identification = identification
+
+
+class TaskInfo(Task):
+    """Task with extra status information on its lifecycle.
+
+    The equivalent of a StatusServer will contain this information
+    in the real implementation."""
+    def __init__(self, identification):
+        super(TaskInfo, self).__init__(identification)
+        self._status = None
+        self._timeout = None
+
+    @property
+    def status(self):
+        return self._status
+
+    @status.setter
+    def status(self, status):
+        self._status = status
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    @timeout.setter
+    def timeout(self, timeout):
+        self._timeout = timeout
+
+    def __repr__(self):
+        if self._status is None:
+            return '%s' % self._identification
+        else:
+            return '%s (%s)' % (self._identification,
+                                self.status)
+
+
+class TaskLifeCycle:
```

Comment:
I don't like much of `TaskLifeCyle` as a class name here. 

Maybe `TaskQueuee` or another alternative would be better.

Reply:
Or `TaskSchedulerState`... just kidding! I'll consider your or other suggestions.

Reply:
`TaskStateMachine`